File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int n = colors.size();
        if(n <= 2) return false;
        
        int alice = 0, bob = 0;
        
        for(int i = 1; i+1 < n; i++) {
            if(colors[i-1] == colors[i] && colors[i] == colors[i+1]) {
                if(colors[i] == 'A') {
                    alice++;
                } else {
                    bob++;
                }
            }
        }
        
        return alice > bob;
    }
};

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.go
func winnerOfGame(colors string) bool {
    n := len(colors)
    if n <= 2 { return false }
        
    alice, bob := 0, 0

    for i := 1; i+1 < n; i++ {
        if (colors[i-1] == colors[i]) && (colors[i] == colors[i+1]) {
            if colors[i] == 'A' {
                alice++;
            } else {
                bob++;
            }
        }
    }

    return alice > bob
}